### PR TITLE
Add a section about trailing underscore variables

### PR DESCRIPTION
**Pull request by rrosenblum**

### CODE REVIEW

**Comment by aredondo** [comment]
Why prefixing `foo` with the splat operator? If `foo` is an array, don't the following two assignments work equally:

    *a, b, _ = *foo
    *a, b, _ = foo

**Comment by bbatsov** [comment]
> If foo is an array, don't the following two assignments work equally:

There are other splattable objects, for this won't be the case.

File: README.md [hunk 1]
```diff
@@ -708,6 +708,28 @@ Translations of the guide are available in the following languages:
   a = *(1..3)
   ```
 
+* <a name="trailing-underscore-variables"></a>
+  Avoid the use of unnecessary trailing underscore variables during
+  parallel assignment. Trailing underscore variables are necessary
+  when there is a splat variable defined in the argument chain, and
```
Comment (bbatsov) [inline README.md]: in the argument chain -> on the left side of the assignment (sounds better to me)

**Comment by rrosenblum** [comment]
Updated, let me know if this sounds better.

**Comment by 0x01f7** [comment]
It's really hard to understand, since we know nothing about **foo**, so I open an [issue](https://github.com/bbatsov/ruby-style-guide/issues/504), colud have a look?

The section is about **unnecessary trailing underscore variables**, why not give a meaning about **foo**? And I still cannot agree the meaning about **unnecessary trailing underscore variable** as I asked in Q4.

Sorry for my poor English.

**Comment by rrosenblum** [comment]
You make an excellent point. Take a look at #506. I have attempted to make the variables more meaningful and provide a bit of reasoning.

**Pull request merged**